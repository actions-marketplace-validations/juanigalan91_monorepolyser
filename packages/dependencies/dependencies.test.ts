import * as dependencies from '.';

jest.mock('./utils', () => {
  return {
    getWorkingDirectory: () => 'root',
    getWorkspaces: () => ['packages/**'],
    getRootPackageJson: () => {
      return {
        name: '@module/root',
        version: '1.0.0',
        dependencies: {
          jest: '26.8.3',
        },
      };
    },
  };
});

jest.mock('glob', () => {
  return {
    sync: () => ['@module/search', '@module/home', '@module/admin'],
  };
});

jest.mock(
  'root/@module/search',
  () => {
    return {
      name: '@module/search',
      version: '1.0.0',
      dependencies: {
        react: '16.8.3',
      },
    };
  },
  { virtual: true }
);

jest.mock(
  'root/@module/home',
  () => {
    return {
      name: '@module/home',
      version: '1.0.0',
      dependencies: {
        react: '16.8.3',
        lodash: '0.19.0',
      },
    };
  },
  { virtual: true }
);

jest.mock(
  'root/@module/admin',
  () => {
    return {
      name: '@module/admin',
      version: '1.0.0',
      dependencies: {
        react: '16.8.3',
      },
    };
  },
  { virtual: true }
);

describe('Dependencies', () => {
  it('returns the workspace packages grouped by package name', () => {
    const project = dependencies.getPackagesFromWorkspaces();

    expect(project).toMatchSnapshot();
    expect(project.packages).toBeDefined();
    expect(project.workspaces).toBeDefined();

    expect(Object.keys(project.packages)).toHaveLength(4);
    expect(project.packages['@module/ljkslkdflkdfslkljdfglhjdgf']).toBeDefined();
    expect(project.packages['@module/home']).toBeDefined();
    expect(project.packages['@module/root']).toBeDefined();
    expect(project.packages['@module/search']).toBeDefined();
  });
});
