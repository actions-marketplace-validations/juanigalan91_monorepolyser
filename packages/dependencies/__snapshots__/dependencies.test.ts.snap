// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dependencies returns the workspace packages grouped by package name 1`] = `
Object {
  "packages": Object {
    "@module/admin": Object {
      "dependencies": Object {
        "react": "16.8.3",
      },
      "name": "@module/admin",
      "version": "1.0.0",
    },
    "@module/home": Object {
      "dependencies": Object {
        "lodash": "0.19.0",
        "react": "16.8.3",
      },
      "name": "@module/home",
      "version": "1.0.0",
    },
    "@module/root": Object {
      "dependencies": Object {
        "jest": "26.8.3",
      },
      "devDependencies": undefined,
      "name": "@module/root",
      "version": "1.0.0",
    },
    "@module/search": Object {
      "dependencies": Object {
        "react": "16.8.3",
      },
      "name": "@module/search",
      "version": "1.0.0",
    },
  },
  "workspaces": Array [
    "packages/**",
  ],
}
`;

exports[`Dependencies returns the workspace packages grouped by package name but it ignores the root package 1`] = `
Object {
  "packages": Object {
    "@module/admin": Object {
      "dependencies": Object {
        "react": "16.8.3",
      },
      "name": "@module/admin",
      "version": "1.0.0",
    },
    "@module/home": Object {
      "dependencies": Object {
        "lodash": "0.19.0",
        "react": "16.8.3",
      },
      "name": "@module/home",
      "version": "1.0.0",
    },
    "@module/search": Object {
      "dependencies": Object {
        "react": "16.8.3",
      },
      "name": "@module/search",
      "version": "1.0.0",
    },
  },
  "workspaces": Array [
    "packages/**",
  ],
}
`;
